src/ArrayHelpers/ArrayHelper.php

```diff
@@ -0,0 +1,67 @@
+<?php
+
+namespace Ingenerator\PHPUtils\ArrayHelpers;
+
+
+use function array_filter;
+use function array_merge;
+use function is_array;
+use function is_null;
+use function is_scalar;
+use function ksort;
+
+class ArrayHelper
+{
+    /**
+     * Recursively remove empty values [], ""
+     * Spares values which evaluate to FALSE boolean FALSE and int 0
+     */
+    public static function removeEmptyValues(array $arr): array
+    {
+        foreach ($arr as &$value) {
+            if (is_array($value)) {
+                $value = self::removeEmptyValues($value);
+            }
+        }
+
+        return array_filter($arr, static fn($item) => ($item === FALSE) || ($item === 0) || ! empty($item));
+    }
+
+    /**
+     * Recursively sorts an array by alpha scalar keys then alpha composite keys
+     */
+    public static function alphaSortKeysScalarComposite(array $arr): array
+    {
+        $scalars    = [];
+        $composites = [];
+
+        foreach ($arr as $k => $v) {
+            if (is_scalar($v) || is_null($v)) {
+                $scalars[$k] = $v;
+            } else {
+                $composites[$k] = $v;
+            }
+        }
+
+        ksort($scalars);
+        ksort($composites);
+
+        foreach ($composites as $key => $array) {
+            $composites[$key] = self::alphaSortKeysScalarComposite($array);
+        }
+
+        return array_merge($scalars, $composites);
+    }
+
+    public static function alphaSortKeysRecursively(array $arr): array
```

Comment:
Also probably better in AssociativeArrayUtils?

Reply:
Same as above